packages/meditrak-server/src/routes/importSurveys/Validator/ConfigValidator/ConditionConfigValidator.js

```diff
@@ -3,27 +3,101 @@
-import { JsonFieldValidator } from '../JsonFieldValidator';
+import { constructIsNotPresentOr, hasContent } from '@tupaia/utils';
+import { CalculatedConfigValidator } from './CalculatedConfigValidator';
 import { splitStringOn, splitStringOnComma, getExpressionQuestionCodes } from '../../../utilities';
 
-export class ConditionConfigValidator extends JsonFieldValidator {
-  constructor(questions, models) {
-    super(questions);
-    this.models = models;
-  }
-
-  static fieldName = 'config';
-
+export class ConditionConfigValidator extends CalculatedConfigValidator {
   getFieldValidators(rowIndex) {
     const conditionsPointToOtherQuestions = this.constructConditionsPointToOtherQuestions(rowIndex);
+
+    return {
+      conditions: [hasContent, constructIsNotPresentOr(conditionsPointToOtherQuestions)],
+      defaultValues: this.getDefaultValuesValidators(rowIndex),
+    };
+  }
+
+  getDefaultValuesValidators(rowIndex) {
+    const existIfQuestionsInFormulaAreOptional = this.hasContentIfQuestionsInFormulaAreOptional(
+      rowIndex,
+    );
     const defaultValuesPointToOtherQuestions = this.constructDefaultValuesPointToOtherQuestions(
       rowIndex,
     );
+    const defaultValuesProvidedForAllOptionalQuestions = this.constructDefaultValuesProvidedForAllOptionalQuestions(
+      rowIndex,
+    );
 
-    return {
-      conditions: [constructIsNotPresentOr(conditionsPointToOtherQuestions)],
-      defaultValues: [constructIsNotPresentOr(defaultValuesPointToOtherQuestions)],
+    return [
+      existIfQuestionsInFormulaAreOptional,
+      constructIsNotPresentOr(defaultValuesPointToOtherQuestions),
+      constructIsNotPresentOr(defaultValuesProvidedForAllOptionalQuestions),
+    ];
+  }
+
+  /**
+   * When any questions used in the formula are optional, users need to provide 'defaultValues'
+   * so that if the optional question is not answered, a default value can be used.
+   * @param {*} rowIndex
+   */
+  hasContentIfQuestionsInFormulaAreOptional(rowIndex) {
+    return (value, object, key) => {
+      const { conditions: conditionsString } = object;
+      const conditions = splitStringOnComma(conditionsString);
+
+      for (const condition of conditions) {
+        const [targetValue, expression] = splitStringOn(condition, ':');
+        this.assertHasContentIfQuestionsInFormulaAreOptional(
+          expression,
+          object,
+          value,
+          key,
+          rowIndex,
+        );
+      }
+      return true;
+    };
+  }
+
+  /**
+   * Validate if, for any optional questions, there are corresponding default values so that if optional questions are not answered, default values can be populated.
+   * @param {*} rowIndex
+   */
+  constructDefaultValuesProvidedForAllOptionalQuestions(rowIndex) {
+    return (value, object) => {
+      const { conditions } = object;
+      const defaultValues = splitStringOnComma(value);
+      const resultValueToExpression = {};
+      const resultValueToCodes = {};
```

Comment:
I would sit each of these declarations directly above the for loop that populates them